### DISCUSSION TITLE
[RFC] turbo-ignore doesn't provide a way to specify a commit to compare

### DISCUSSION THREAD

**Discussion by denexapp** [vercel/turborepo#7465] [Ideas]
### Which project is this feature idea for?

Turborepo

### Backstory

This is a continuation of our conversation with @anthonyshew on the turbo discord server: https://discord.com/channels/818588653005176832/1108910806269169785

It has led to some improvements in the docs, namely https://github.com/vercel/turbo/pull/5039 and https://github.com/vercel/turbo/pull/5041, which has resolved some inconsistency between the docs and the actual behaviour, but it doesn't solve the problem.

### The problem

Turborepo docs have a page called [Skipping tasks in CI](https://turbo.build/repo/docs/core-concepts/monorepos/skipping-tasks). The docs tell about a simple CLI tool, `turbo-ignore`, that uses `turbo` under the hood. This tool returns status code 1 if a workspace have changes and 0 otherwise. The docs page suggests that you should use it in CI.

In a CI pipeline that starts when a new commit gets pushed, you want to compare this commit with a commit of the latest sucessful deployment. The docs page mentioned before implies that you can do that by showing the following command as the example of a command that gets generated under the hood, when `turbo-ignore` interacts with `turbo`. Note the specified commit sha:
```
npx turbo run build --filter=@example/web...3c8387ffd98b751305fe3f0284befdd00cbd4610 --dry=json
```

On Vercel, it happens automatically - `turbo-ignore` adapts if it sees Vercel specific environment variables. On the other hand, in your own CI you're supposed to provide it by yourself. The problem is that there's no way to provide it, which makes `turbo-ignore` basically useless outside of Vercel.

### Describe the solution you'd like

I don't have an exact solution, but there should be a way to provide commit sha to compare to. One idea might be to introduce a new flag that accepts it.

### Describe alternatives you've considered

I tried using the `--fallback` flag, as the docs page suggested to before the mentioned PR's were merged, and it didn't worked.

I ended up copying parts of `turbo-ignore` into my own CI scripts, so I manually call `turbo` with the `--dry` flag and look for the `packages` field in the returned JSON.

**Comment by mehulkar** [comment]
Reading the code, it looks like you _could_ spoof this by [setting some environment variables](https://github.com/vercel/turbo/blob/main/packages/turbo-ignore/src/getComparison.ts), but that isn't ideal. cc @tknickman for suggestions. It seems like a relatively easy script to duplicate in userland, if we aren't intending on turbo-ignore to be usable outside Vercel. If @tknickman is ok with adding a more general purpose feature, I'm sure we'd be open to a pull request!

**Comment by tknickman** [comment]
Yea we can certainly extend this to provide a custom comparison

**Comment by denexapp** [comment]
Do you mind if I make a PR, or do you plan to implement it internally?

**Comment by tknickman** [comment]
You are more than welcome to make a PR!
